README.md

```diff
@@ -1019,6 +1019,255 @@ Voilà! Since we updated `index.html`, you'll need to restart the dev server. Go
+                        mergeFeeds(feed, data)
+                      | _ => data
+                      };
+                    {
+                      ...state,
+                      step: Loaded(Ok(updatedFeed)),
+                      currentPage: page + 1,
+                    };
+                  })
+                };
+              }
+              |> P.resolve
+            );
+        } else {
+          /* Handle non-200 status */
+          ...
+        };
+      })
+  |> ignore;
+};
+```
+
+#### 11.5: Modify render code to include scroll sentinel and handling new state
+
+We’ll add a `sentinelRef` to track the last element of the list, which will
+trigger loading more data when it comes into view.
+
+In the render code, create a new React ref to track the last element of the
+list:
+
+```reason
+    let sentinelRef = React.useRef(Js.Nullable.null);
+```
+
+Also, add the element itself at the end of the `switch`:
+
+```reason
+  {switch (state.step) {
+    | InvalidUsername => <div> {React.string("Invalid username")} </div>
+    | Idle =>
+      <div>
+        {React.string(
+          "Press the \"Enter\" key to confirm the username selection.",
+        )}
+      </div>
+    | Loading(None | Some(Error(_))) =>
+      <div> {React.string("Loading...")} </div>
+    | Loading(Some(Ok(feed))) =>
+      <> {renderFeed(feed)} <div> {React.string("Loading...")} </div> </>
+    | Loaded(Error(msg)) => <div> {React.string(msg)} </div>
+    | Loaded(Ok(feed)) => renderFeed(feed)
+    }}
+  <div ref={ReactDOM.Ref.domRef(sentinelRef)} />
+```
+
+#### 11.6: Update the effect to fetch new data when observer is intersecting
+
+Finally, let's update the effect to fetch more data when the
+`IntersectionObserver` detects that the sentinel element is in the viewport.
+
+```reason
+  React.useEffect1(
+    () => {
+      switch (Username.make(state.username)) {
+      | Error () =>
+        Js.Exn.raiseError("The value of state.username is invalid")
+      | Ok(username) =>
+        switch (Js.Nullable.toOption(sentinelRef.current)) {
+        | None =>
+          ();
+          None;
+        | Some(elem) =>
+          let shouldFetch =
+            switch (state.step) {
+            | Loading(_) => false
+            | Loaded(Ok({entries: [||]})) => false
+            | Idle
+            | InvalidUsername
+            | Loaded(_) => true
+            };
+          let observer =
+            IntersectionObserver.make(entries => {
+              let entry = entries[0];
+              if (entry.isIntersecting && shouldFetch) {
+                fetchFeed(~username, ~page=state.currentPage);
+              };
+            });
+          IntersectionObserver.observe(observer, elem);
+          Some(() => IntersectionObserver.disconnect(observer));
+        }
+      }
+    },
+    [|state|],
+  );
+```
+
+### Step 11 completion check
+
+Restart the dev server and scroll to the bottom of the page in your browser. The
```

Comment:
I don't think it's necessary to restart the dev server

Reply:
https://github.com/ahrefs/react-alicante-workshop/pull/9/commits/e2a987648cb3329d526a679a5fb51026daa906e4.